### ISSUE TITLE
Show corresponding mobile network name everywhere alongside SIM card slot number

### ISSUE THREAD

**Issue by figvam** [FossifyOrg/Phone#162]
### Checklist

- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Phone/issues) or [closed](https://github.com/FossifyOrg/Phone/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Phone/discussions) or [closed](https://github.com/FossifyOrg/Phone/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Phone/blob/master/CONTRIBUTING.md).
- [ ] I optionally [donated](https://fossify.org/donate) to support the Fossify mission.

### Feature description

Show human-readable mobile network name alongside SIM card number, where applicable.

### Why do you want this feature?

For multi-SIM phones, it would be nice to show a human-readable mobile network name wherever a SIM card slot number is shown, for example on the receiving call screen, on the ongoing call screen and in the call history screen.

I think seeing just SIM card slot numbers may be confusing for people who have just set up the phone and are not yet accustomed to SIM slot <-> mobile operator relationship.

**Comment by chesio** [comment]
I would be careful about that - mobile network provider names can be quite long and there is no way to customize them (as far as I know).

**Comment by blobsan** [comment]
I think it would be more straight-forward if the dial buttons would actually use the color and name set for the SIMs in the Android settings. This can be queried with SubscriptionManager.getActiveSubscriptionInfo (API Level >=22).

Btw is there a way to provide a donation or bounty dedicated to certain issues or features?

**Comment by Cwpute** [comment]
> @blobsan : Btw is there a way to provide a donation or bounty dedicated to certain issues or features?

Just publicly announce your upcoming donation here :) you can even donate preemptively to show you trust and support for the team !

(there are several tools to post and track bounties, but last year's treachery of Bountysource left a bitter taste in everybody's mouth, so you're better off just making promises yourself!)

**Comment by omarsoufiane** [comment]
@blobsan Hi I'm the founder of bountyhub,
unlike bountysource we have a straightforward [workflow](https://www.bountyhub.dev/en/docs/create-bounty#create-bounty-workflow) that would prevent problems that occur when dealing with bounties. TLDR: you only pay when a pr is merged, and even if you don't want to pay if a PR is merged we have a dispute resolution system where we manually check if the pr solves the issue.

**Comment by naveensingh** [comment]
We won't do bounties and there is no "donate to prioritize" scheme.

Also, @omarsoufiane your message looks like an ad, please do not advertise.

**Comment by Antiplan** [comment]
I recently had a issue related to this. It came in form of a bill over roughly 4xx euros. The phones these days have that thing called eSIM. Of course in the phone settings I had the correct SIM-Card selected for calls and messages and data, so as a phone-noob I assumed that this selected/preferred sim would be the sim 1 in this dialer app. Well... turns out I was wrong. There is no way to select or even tell which sim you are using to make a call. This is such a nice app, but that design choice is... very questionable, to say the least. Why ignore the settings? Of course I had no plan installed for the second sim, it's for emergency calls only. So all my calls were treated as high priority calls (as should be, from a high priority sim) and I was charged accordingly. I wasn't actually working for most of the month, so there were only a few calls. Got lucky there, I guess.

I was naive to not properly test this app before I used it, but the option to... well... see the sim-cards in the first place and then select/activate it (or/and disable it!) in the settings would be nice. Essential, really. There is no way to tell which of the sim-slots are considered to be sim 1 and sim 2 by the app. To have both sim available by button is kinda nice, to be honest, but not knowing which is which...? Huh.

This was using Android 14 on a Motorola G34 5G.

Basically it would be nice to have a "global default preferred sim-slot"-option in case no option is set. Or at least a way to see which sim is beeing used to make the call. Maybe display the sims caller id instead of simply "1" and "2" or the last 3 or 4 digits of the caller id. Something on those lines...

**Comment by candroid-man** [comment]
As a business owner I need this feature. I use Cheogram/Snickett as my business line for my shop which is called constantly, and it uses the system dialer when a call comes in. I need to know if someone is calling my personal number or if it is a business call so I know how to answer the call. I'm going to have to use something else until this is implemented unfortunately, but I will be keeping an eye on this issue so that hopefully soon I can switch back

**Comment by Cwpute** [comment]
> I'm going to have to use something else until this is implemented unfortunately

Please do share if you find a suitable solution that doesn't require me selling my soul to a tie-wearing devil.
